src/wp-includes/media.php

```diff
@@ -793,6 +810,117 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
 }
 
 /**
+ * Return an array of urls and pixel widths representing sizes of an image for
+ * the purpose of populating a source set when creating responsive image markup.
+ *
+ * @since 4.4.0
+ *
+ * @param  int    $id   Image attachment ID.
+ * @param  string $size Optional. Name of image size. Default value: 'medium'.
+ * @return array|bool  An array of image urls and widths or false on failure.
+ */
+function wp_attachment_img_srcset_array( $id, $size = 'medium' ) {
+	// See which image is being returned and bail if none is found.
+	if ( ! $img = wp_get_attachment_image_src( $id, $size ) ) {
+		return false;
+	}
+
+	// Break image data into url, width, and height.
+	list( $img_url, $img_width, $img_height ) = $img;
+
+	// Bail early If we have no width or if the width set to zero.
+	if ( ! $img_width > 0 ) {
```

Comment:
It would probably be safe to bail if there is no height, or if height was 0px as well, but technically speaking we only need to worry about widths because a) we use the width as the denominator when calculating aspect ratios, and b) we're using `w` descriptors in our `srcset` values.